internal/impl/update.go

```diff
@@ -54,6 +54,10 @@ func (d *Devbox) Update(ctx context.Context, pkgs ...string) error {
 		}
 	}
 
+	if err := d.lockfile.Save(); err != nil {
```

Comment:
One downside to this approach is that if anything breaks while installing packages (which is fairly common due to stuff like insecure packages) then we leave the lockfile in a bad state that doesn't represent reality. This would have happened while updating `nodejs@16`

An alternative approach is to have `local.IsUpToDate` take a lock file instead of reading from disk. This would guarantee not running into this issue again. Similarly, `localLock.Update()` would take a lock file.

Not for this PR, but I think we could improve the local lock file API. Specifically I would make it private and just have the lock struct interact with it. e.g.

`d.lockfile.LocalIsUpToDate()`

And privatize `local.Update` (just have it be called by `lock.Save()` since they are both called together anyway.

This would make all the local stuff private.

Reply:
> An alternative approach is to have local.IsUpToDate take a lock file instead of reading from disk. 

Yeah.. I have an implementation of this, but I prefer to get this in and later think about how to refactor the local lock stuff, since it may have larger implications.